dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -381,9 +383,11 @@ private Map<String, MetadataItem> getMetadataItems( EventQueryParams params,
     }
 
     private void addItemIntoMetadata( final Map<String, MetadataItem> metadataItemMap, final QueryItem item,
-        final boolean includeDetails )
+        final boolean includeDetails, final DisplayProperty displayProperty )
     {
-        final MetadataItem metadataItem = new MetadataItem( item.getItem().getDisplayName(),
+        final MetadataItem metadataItem = new MetadataItem(
```

Comment:
Instead of adding this logic you could use this approach `item.getItem().getDisplayProperty( displayProperty )`

Reply:
done